src/tls13.c

```diff
@@ -2652,9 +2695,32 @@ static int SetupPskKey(WOLFSSL* ssl, PreSharedKey* psk)
         const char* cipherName = NULL;
         byte cipherSuite0 = TLS13_BYTE, cipherSuite = WOLFSSL_DEF_PSK_CIPHER;
         int cipherSuiteFlags = WOLFSSL_CIPHER_SUITE_FLAG_NONE;
-
+        
+        #ifdef OPENSSL_EXTRA
+        const unsigned char* id = NULL;
```

Comment:
Try checking if "ssl->options.session_psk_cb != NULL".
Then do all this work.

Reply:
Checking if "ssl->options.session_psk_cb != NULL" first, then do the following work